reana_workflow_controller/consumer.py

```diff
@@ -144,16 +145,28 @@ def _update_workflow_status(workflow, status, logs):
             _update_commit_status(workflow, status)
 
         if status not in ALIVE_STATUSES:
+            workflow.run_finished_at = datetime.now()
+            workflow.logs = workflow.logs or ""
+
             try:
-                workflow.run_finished_at = datetime.now()
-                if RunStatus.should_cleanup_job(status):
-                    _delete_workflow_engine_pod(workflow)
-            except REANAWorkflowControllerError:
+                workflow_engine_logs = _get_workflow_engine_pod_logs(workflow)
+                workflow.logs += workflow_engine_logs + "\n"
+            except REANAWorkflowControllerError as exception:
                 logging.error(
-                    f"Could not clean up workflow engine for workflow {workflow.id_}"
+                    f"Could not fetch workflow engine pod logs for workflow {workflow.id_}."
+                    f" Error: {exception}"
                 )
                 workflow.logs += "Workflow engine logs could not be retrieved.\n"
 
+            if RunStatus.should_cleanup_job(status):
+                try:
+                    _delete_workflow_job(workflow)
+                except REANAWorkflowControllerError as exception:
```

Comment:
When the `_delete_workflow_job` function fails and raises `REANAWorkflowControllerError`exception, we log the `Could not clean up workflow job for workflow...` message. And, when `REANAWorkflowControllerError` is raised in `_delete_workflow_job`, we have a message `Workflow engine pod could not be deleted...`. Those messages are basically the same. I feel like it is a bit overkill. What about just logging the exception itself? LIke:

```python
try:
  _delete_workflow_job(workflow)
except REANAWorkflowControllerError as exception:
  logging.error(exception)
```

Reply:
Since you created dedicated functions `_get_workflow_engine_pod_logs()` and `_delete_workflow_job()` with `try except` blocks inside I think it's not worth to have it in `_update_workflow_status()` as well. All the logging could be done in dedicated functions and the main `_update_workflow_status()` would be simplified a bit, WDYT?

Reply:
In the `except` block for workflow engine logs in `_update_workflow_status`, I have this part:

```python
try:
  ...
except REANAWorkflowControllerError as exception:
  ...
  workflow.logs += "Workflow engine logs could not be retrieved.\n"
```
If I will hide exception handling inside `_get_workflow_engine_pod_logs,  I will not be able to add to `workflow.logs`. I would prefer to still raise an exception inside the `_delete_workflow_job()` and `_get_workflow_engine_pod_logs()`. And handle them in `_update_workflow_status()`.

Reply:
Ohh... It is merged already. I didn't add this change 😢

Reply:
From your last comment I understood there is nothing to change. You were referring to the first comment perhaps? My bad, I guess, sorry

Reply:
It is okay. It was not clear from my side that I want to add more stuff.

Would be good to adjust how we re-request review so things like this are not happening.